actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/OrgBulkUploadBackGroundJobActor.java

```diff
@@ -0,0 +1,204 @@
+package org.sunbird.learner.actors.bulkupload;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.IOException;
+import java.sql.Timestamp;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.actor.router.ActorConfig;
+import org.sunbird.actorutil.org.OrganisationClient;
+import org.sunbird.actorutil.org.impl.OrganisationClientImpl;
+import org.sunbird.common.Constants;
+import org.sunbird.common.models.util.*;
+import org.sunbird.common.request.ExecutionContext;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.learner.actors.bulkupload.model.BulkUploadProcess;
+import org.sunbird.learner.actors.bulkupload.model.BulkUploadProcessTask;
+import org.sunbird.learner.util.Util;
+
+/** Created by rajatgupta on 15/10/18. */
+@ActorConfig(
+  tasks = {"orgBulkUploadBackground"},
+  asyncTasks = {}
+)
+public class OrgBulkUploadBackGroundJobActor extends BaseBulkUploadActor {
+  OrganisationClient orgClient = new OrganisationClientImpl();
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    String operation = request.getOperation();
+    Util.initializeContext(request, TelemetryEnvKey.ORGANISATION);
+    ExecutionContext.setRequestId(request.getRequestId());
+    if (operation.equalsIgnoreCase("orgBulkUploadBackground")) {
+      bulkOrgUpload(request);
+    } else {
+      ProjectLogger.log(operation + ": unsupported message");
+    }
+  }
+
+  private void bulkOrgUpload(Request request) {
+    String processId = (String) request.get(JsonKey.PROCESS_ID);
+    ProjectLogger.log(
+        "OrgBulkUploadBackGroundJobActor: bulkOrgUpload called with process ID " + processId,
+        LoggerEnum.INFO);
+    BulkUploadProcess bulkUploadProcess = bulkUploadDao.read(processId);
+    if (null == bulkUploadProcess) {
+      ProjectLogger.log("Process Id does not exist : " + processId, LoggerEnum.ERROR);
+      return;
+    }
+    Integer status = bulkUploadProcess.getStatus();
+    if (!(status == (ProjectUtil.BulkProcessStatus.COMPLETED.getValue())
+        || status == (ProjectUtil.BulkProcessStatus.INTERRUPT.getValue()))) {
+      try {
+        processOrgBulkUpload(bulkUploadProcess);
+      } catch (Exception ex) {
+        bulkUploadProcess.setStatus(ProjectUtil.BulkProcessStatus.FAILED.getValue());
+        bulkUploadProcess.setFailureResult(ex.getMessage());
+        bulkUploadDao.update(bulkUploadProcess);
+        ProjectLogger.log("Org Bulk BackGroundJob failed processId - " + processId, ex);
+      }
+    }
+    bulkUploadProcess.setStatus(ProjectUtil.BulkProcessStatus.COMPLETED.getValue());
+    bulkUploadDao.update(bulkUploadProcess);
+  }
+
+  private void processOrgBulkUpload(BulkUploadProcess bulkUploadProcess)
```

Comment:
Can we push this to base class and reuse in OrgBulkUploadBackGroundJobActor and LocationBulkUploadBackGroundJobActor? The derived class can override a method or use lambda function.